src/agent/debuglet.ts

```diff
@@ -115,6 +115,8 @@ export class Debuglet extends EventEmitter {
   private project_: string|null;
   private controller_: Controller;
   private completedBreakpointMap_: {[key: string]: boolean};
+  private promiseInitialized_: boolean;
+  private promiseResolve_: (value?: void|PromiseLike<void>|undefined) => void;
```

Comment:
@gaofanmichael The way Promises are being used, it looks like an event emission might be more suitable (anyone who needs to wait for a Promise returned by `initializationPromise` to resolve can just as easily listen for an `"initialized"` event.) Is there a reason in particular we need to use a Promise?

Reply:
Do you mean as below? 
debug.initializePromise();
process.on('breakpointfetched', () => {})